### ISSUE TITLE
ability to limit results

### ISSUE THREAD

**Issue by pmaria** [carml/carml-jar#9]
For mappings that lead to large results it is interesting to be able to limit that result, e.g. during testing.

## Introduce `limit` option
Introduce a new `--limit` option that limits the nr of triples outputted.

This can be easily achieved by applying [Flux#take](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#take-long-) before writing the output.

**Comment by tobiasschweizer** [comment]
Hi @pmaria,

I tried the latest version (including #8) and now the resulting RDF is directly written to stdout without any logging. This is nice!

I noticed that when I map quite a big dataset not the complete result is written out. Is there a default limit or something like that? 

Thanks for your feedback.

**Comment by tobiasschweizer** [comment]
This is just an idea but could it be related to `Outputstream` / System.out? https://github.com/carml/carml-jar/blob/734699b573216f446fa107f330de71ab401abc25/carml-runner/src/main/java/io/carml/runner/CarmlMapCommand.java#L155

However, it also happens when I set an output path.

**Comment by pmaria** [comment]
Hi @tobiasschweizer, there shouldn't be any default limit. Do you have something I can reproduce to see what's going on?

**Comment by tobiasschweizer** [comment]
Sure :-)

data file: [820970.xml.txt](https://github.com/carml/carml-jar/files/9479826/820970.xml.txt)

mapping:  [mapping.cordis.ttl.txt](https://github.com/carml/carml-jar/files/9479819/mapping.cordis.ttl.txt)

When I map this using ` java -jar ../carml-jar-1.0.0-SNAPSHOT-0.4.1.jar map -m mapping.cordis_rmlmapper.ttl -r . ` I do not get all the expected triples. For example, `schema:ResearchProject` and `schema:MonetaryGrant` are missing. I get 1054 triples in total.

I tried this with [RML Mapper](https://github.com/RMLio/rmlmapper-java/releases/download/v6.0.0/rmlmapper-6.0.0-r363-all.jar) using ``and there I get 1359 triples (including `schema:ResearchProject` and `schema:MonetaryGrant`). 

I played a b it around with the mapping: If I remove the instructions for organization and scholarly article, the project and the grant are mapped. Maybe this is another issue. Could it be related to the different logical sources? I'd grateful for any hint.
